packages/forms/src/UIForm/fields/ListView/ListView.component.js

```diff
@@ -0,0 +1,222 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import keycode from 'keycode';
+import ListView from '@talend/react-components/lib/ListView';
+
+import { getItemsProps, initItems, updateItems } from './ListView.utils';
+import FieldTemplate from '../FieldTemplate';
+
+const DISPLAY_MODE_DEFAULT = 'DISPLAY_MODE_DEFAULT';
+const DISPLAY_MODE_SEARCH = 'DISPLAY_MODE_SEARCH';
+const DEFAULT_ITEM_HEIGHT = 33;
+
+class ListViewWidget extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.defaultHeaderActions = [
+			{
+				icon: 'talend-search',
+				id: `${props.id}-search`,
+				label: 'Search for specific values',
+				onClick: this.switchToSearchMode.bind(this),
+			},
+		];
+		this.searchHeaderActions = [
+			{
+				label: 'Abort',
+				icon: 'talend-cross',
+				id: 'abort',
+				onClick: this.switchToDefaultMode.bind(this),
+			},
+		];
+
+		this.state = {
+			...initItems(props.schema, props.value, null, this.onToggleItem.bind(this)),
+			getItemHeight: () => DEFAULT_ITEM_HEIGHT,
+			headerDefault: this.defaultHeaderActions,
+			onAddKeyDown: this.onInputKeyDown.bind(this),
+			onInputChange: this.onInputChange.bind(this),
+			onToggleAll: this.onToggleAll.bind(this),
+		};
+	}
+
+	/**
+	 * On new schema : we redefine the items
+	 * On new value : we update the check status
+	 * @param { Object } schema The new mergedSchema
+	 * @param { Array } value The new value
+	 */
+	componentWillReceiveProps({ schema, value }) {
+		if (schema !== this.props.schema) {
+			this.setState(oldState =>
+				initItems(schema, value, oldState.searchCriteria, this.onToggleItem.bind(this)),
+			);
+		} else if (value !== this.props.value) {
+			this.setState(oldState =>
+				updateItems(oldState.items, value, oldState.searchCriteria)
+			);
+		}
+	}
+
+	/**
+	 * Propagate a ListView value change
+	 * @param { Object } event The event that triggered the change
+	 * @param { Array } newValue The new Value
+	 */
+	onChange(event, newValue) {
+		const value = newValue.length ? newValue : undefined;
+		const payload = { schema: this.props.schema, value };
+		this.props.onChange(event, payload);
+		this.props.onFinish(event, payload);
+	}
+
+	/**
+	 * Search input change
+	 * @param { Object } event The change event
+	 * @param { Array } value
+	 */
+	onInputChange(event, { value }) {
+		if (this.timerSearch) {
```

Comment:
You could delete this test. `clearTimeout(undefined);` is not a problem, so this method could be simplified : 

```javascript
clearTimeout(this.timerSearch);
this.timerSearch = setTimeout(() => {
	this.setState(getItemsProps(this.state.items, value));
}, 400);
```